### PR TITLE
Set required_irql in sockops and sockaddr program information

### PR DESCRIPTION
**Pull request by Alan-Jowett**
## Description

The sockops and sockaddr program information are missing the required_irql fields. This causes bpf_prog_test_run_opts to invoke them at IRQL == PASSIVE_LEVEL instead of IRQL == DISPATCH_LEVEL.

## Testing

CI/CD

## Documentation

No.

## Installation

No.

### CODE REVIEW

**Review by mtfriesen — DISMISSED** [review]

**Review by dv-msft — DISMISSED** [review]

File: docs/eBpfExtensions.md [hunk 1]
```diff
@@ -73,7 +73,7 @@ functions that override the global helper functions provided by the eBPF runtime
 structure from provided data and context buffers.
 * `context_destroy`: Pointer to `ebpf_program_context_destroy_t` function that destroys a program type specific
 context structure and populates the returned data and context buffers.
-* `required_irql`: IRQL that the eBPF program runs at.
+* `required_irql`: IRQL that the eBPF program runs at when being invoked via bpf_prog_test_run_opts.
```
Comment (dthaler) [inline docs/eBpfExtensions.md:76]:
```suggestion
* `required_irql`: IRQL at which the eBPF program is invoked by bpf_prog_test_run_opts.
```

Comment (dthaler) [reply docs/eBpfExtensions.md:76]: So does this mean the IRQL used by bpf_prog_test_run_opts, and the IRQL from a real invocation can be different? If so, the label "required_irql' seems wrong to me.

**Review by mtfriesen — APPROVED** [review]

File: docs/eBpfExtensions.md (hunk 1 above)
Comment (Alan-Jowett) [reply docs/eBpfExtensions.md]: Agreed. Longer term there is a need to have the bpf_prog_test_run_opts specify what IRQL to run the BPF program at and for the extension to be able to validate this, which will be addressed by https://github.com/microsoft/ebpf-for-windows/issues/3157, but that is a larger change. The goal of this is to unblock the BPF performance testing so we can detect regressions.

Before #3157 can be implemented, there needs to be a decision on:
1) Do we need to update the version of the interface for pre-release eBPF core or is a breaking changer acceptable.
2) How can this be passed this via the libbpf API in a way that maintains compatability.

Comment (dthaler) [reply docs/eBpfExtensions.md]: Agreed that the label is wrong?  If so, why not change it now?

**Review by dv-msft — APPROVED** [review]

File: docs/eBpfExtensions.md (hunk 1 above)
Comment (dthaler) [reply docs/eBpfExtensions.md]: Ok, I take it changing the label would be a breaking change, yes?

**Review by dthaler — APPROVED** [review]

**Pull request merged**